correction of IANA registry names (#85)
openpgp-pqc/draft-openpgp-pqc: draft-ietf-openpgp-pqc.md

# IANA Considerations

IANA is requested to add the following registries to the `OpenPGP`

> registry group at https://www.iana.org/assignments/openpgp/openpgp.xhtml:

Reviewer: i would drop the trailing `/openpgp.xhtml` -- or maybe drop the whole URL, since we want the updates to happen to the registry itself, not just the html variant.

Author: > i would drop the trailing `/openpgp.xhtml`

Fixed now.

- Registry name: `OpenPGP SLH-DSA-SHA2 parameters`

  Registration procedure: SPECIFICATION REQUIRED [RFC8126]

  The registry contains the values defined in {{slhdsa-param-sha2}} in this document.

- Registry name: `OpenPGP SLH-DSA-SHAKE parameters`

  Registration procedure: SPECIFICATION REQUIRED [RFC8126]

  The registry contains the values defined in {{slhdsa-param-shake}} in this document.

Furthermore, IANA is requested to add the algorithm IDs defined in {{kem-alg-specs}} and

> {{sig-alg-specs}} to the existing registry `OpenPGP Public Key Algorithms`.

Reviewer: the "OpenPGP Public Key Algorithms" table has the following columns:

ID, Algorithm, Public Key Format, Secret Key Format, Signature Format, PKESK Format, Reference

The tables linked here don't tell IANA how to populate all the columns.

Author: Added a table with all the information.

# IANA Considerations

IANA is requested to add the following registries to the `OpenPGP`
registry group at https://www.iana.org/assignments/openpgp:

- Registry name: `OpenPGP SLH-DSA-SHA2 parameters`

  Registration procedure: SPECIFICATION REQUIRED [RFC8126]

  The registry contains the values defined in {{slhdsa-param-sha2}} in this document.

- Registry name: `OpenPGP SLH-DSA-SHAKE parameters`

  Registration procedure: SPECIFICATION REQUIRED [RFC8126]

  The registry contains the values defined in {{slhdsa-param-shake}} in this document.

> Furthermore, IANA is requested to add the algorithm IDs defined in {{iana-pubkey-algos}} to the existing registry `OpenPGP Public Key Algorithms`. The bracketed fields for IDs 105 and 106 denote fields the presence of which is conditional.

Reviewer 2: ```suggestion
Furthermore, IANA is requested to add the algorithm IDs defined in {{iana-pubkey-algos}} to the existing registry `OpenPGP Public Key Algorithms`. Brackets denote fields the presence of which is conditional.
```

Author: I corrected the sentence in yet another way.

ID     | Algorithm           | Public Key Format                                                                                                  | Secret Key Format                                                                                                  | Signature Format                                                                                                  | PKESK Format                                                                                                                                                                                    | Reference
---  : | -----               | ---------:                                                                                                         | --------:                                                                                                          | --------:                                                                                                         | -----:                                                                                                                                                                                          | -----:

> TBD    | ML-KEM-768 + X25519 | 32 octets X25519 public key ({{tab-ecdh-cfrg-artifacts}}), 1184 octets ML-KEM public key ({{tab-mlkem-artifacts}}) | 32 octets X25519 secret key ({{tab-ecdh-cfrg-artifacts}}), 2400 octets ML-KEM secret-key ({{tab-mlkem-artifacts}}) | N/A                                                                                                               | 32 octets ECC ciphertext, 1088 octets ML-KEM ciphertext \[, 1 octet algorithm ID in case of v3 PKESK\], 1 octet length field of value `n`, `n` octets wrapped session key ({{ecc-mlkem-pkesk}}) | {{ecc-mlkem}}

Reviewer 3: ```suggestion
TBD    | ML-KEM-768 + X25519 | 32 octets X25519 public key ({{tab-ecdh-cfrg-artifacts}}), 1184 octets ML-KEM-768 public key ({{tab-mlkem-artifacts}}) | 32 octets X25519 secret key ({{tab-ecdh-cfrg-artifacts}}), 2400 octets ML-KEM-768 secret-key ({{tab-mlkem-artifacts}}) | N/A                                                                                                               | 32 octets ECC ciphertext, 1088 octets ML-KEM-768 ciphertext \[, 1 octet algorithm ID in case of v3 PKESK\], 1 octet length field of value `n`, `n` octets wrapped session key ({{ecc-mlkem-pkesk}}) | {{ecc-mlkem}}
```

Reviewer 3: Complete reference to ML-KEM parameter set like you did it for ML-DSA.

> TBD    | ML-KEM-1024 + X448  | 56 octets X448 public key ({{tab-ecdh-cfrg-artifacts}}), 1568  octets ML-KEM public key ({{tab-mlkem-artifacts}})  | 56 octets X448 secret key ({{tab-ecdh-cfrg-artifacts}}), 3168 octets ML-KEM secret-key ({{tab-mlkem-artifacts}})   | N/A                                                                                                               | 56 octets ECC ciphertext, 1568 octets ML-KEM ciphertext \[, 1 octet algorithm ID in case of v3 PKESK\], 1 octet length field of value `n`, `n` octets wrapped session key ({{ecc-mlkem-pkesk}}) | {{ecc-mlkem}}

Reviewer 3: ```suggestion
TBD    | ML-KEM-1024 + X448  | 56 octets X448 public key ({{tab-ecdh-cfrg-artifacts}}), 1568  octets ML-KEM-1024 public key ({{tab-mlkem-artifacts}})  | 56 octets X448 secret key ({{tab-ecdh-cfrg-artifacts}}), 3168 octets ML-KEM-1024 secret-key ({{tab-mlkem-artifacts}})   | N/A                                                                                                               | 56 octets X448 ciphertext, 1568 octets ML-KEM-1024 ciphertext \[, 1 octet algorithm ID in case of v3 PKESK\], 1 octet length field of value `n`, `n` octets wrapped session key ({{ecc-mlkem-pkesk}}) | {{ecc-mlkem}}
```

Reviewer 3: As above

TBD    | ML-DSA-65 + Ed25519 | 32 octets Ed25519 public key ({{tab-eddsa-artifacts}}), 1952 octets ML-DSA-65 public key ({{tab-mldsa-artifacts}}) | 32 octets Ed25519 secret key ({{tab-eddsa-artifacts}}), 4000  octets ML-DSA-65 secret ({{tab-mldsa-artifacts}})    | 64 octets Ed25519 signature ({{tab-eddsa-artifacts}}), 3293 octets ML-DSA-65 signature ({{tab-mldsa-artifacts}})  | N/A                                                                                                                                                                                             | {{ecc-mldsa}}

> TBD    | ML-DSA-87 + Ed448   | 57 octets Ed448 public key ({{tab-eddsa-artifacts}}),  2592 octets ML-DSA-87 public key ({{tab-mldsa-artifacts}})  | 57 octets Ed448 secret key ({{tab-eddsa-artifacts}}), 4864 octets ML-DSA-87 secret ({{tab-mldsa-artifacts}})       | 114 octets Ed25519 signature ({{tab-eddsa-artifacts}}), 4595 octets ML-DSA-87 signature ({{tab-mldsa-artifacts}}) | N/A                                                                                                                                                                                             | {{ecc-mldsa}}

Reviewer 3: ```suggestion
TBD    | ML-DSA-87 + Ed448   | 57 octets Ed448 public key ({{tab-eddsa-artifacts}}),  2592 octets ML-DSA-87 public key ({{tab-mldsa-artifacts}})  | 57 octets Ed448 secret key ({{tab-eddsa-artifacts}}), 4864 octets ML-DSA-87 secret ({{tab-mldsa-artifacts}})       | 114 octets Ed448 signature ({{tab-eddsa-artifacts}}), 4595 octets ML-DSA-87 signature ({{tab-mldsa-artifacts}}) | N/A                                                                                                                                                                                             | {{ecc-mldsa}}
```

Reviewer 3: Typo

> TBD    | SLH-DSA-SHA2        | 1 octet parameter ID, per parameter fixed-length octet string ({{slhdsa-artifact-lengths}})                        | per parameter fixed-length octet string ({{slhdsa-artifact-lengths}})                                              | 1 octet parameter ID, per parameter fixed-length octet string ({{slhdsa-artifact-lengths}})                       | N/A                                                                                                                                                                                             | {{slhdsa}}

Reviewer 3: Is SLH-DSA-SHAKE missing in the table?

Author: Yes, indeed. Now added.
